### PR TITLE
Upgrade to TS version of ethers

### PR DESCRIPTION
**Pull request by LogvinovLeon**
## Description

Ethers.js releases the new version with typescript support. This PR upgrades us to that version.
This version is still in beta, but a couple of people at ETHBerlin convinced me that it's solid already.
We also stop using our custom fork for revert reasons as it's now merged upstream.

## Testing instructions

CI build passes

## Types of changes

Dependency upgrade

## Checklist:

*   [x] Prefix PR title with `[WIP]` if necessary.
*   [x] Prefix PR title with bracketed package name(s) corresponding to the changed package(s). For example: `[sol-cov] Fixed bug`.
*   [x] Add tests to cover changes as needed.
*   [x] Update documentation as needed.
*   [x] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

**Comment by albrow** [comment]
@LogvinovLeon please rebase off of development and then CI should pass 🤞

File: packages/0x.js/package.json [hunk 1]
```diff
@@ -84,7 +84,7 @@
         "@0xproject/utils": "^1.0.11",
         "@0xproject/web3-wrapper": "^3.0.1",
         "ethereum-types": "^1.0.8",
-        "ethers": "3.0.22",
+        "ethers": "^4.0.0-beta.14",
```
Comment (fabioberger) [inline packages/0x.js/package.json]: I'm sure Ethers will introduce breaking changes during their beta that will get caught by `^`. Can we leave it out until they are out of beta?

File: packages/base-contract/package.json [hunk 2]
```diff
@@ -44,7 +44,7 @@
         "@0xproject/utils": "^1.0.11",
         "@0xproject/web3-wrapper": "^3.0.1",
         "ethereum-types": "^1.0.8",
-        "ethers": "3.0.22",
+        "ethers": "^4.0.0-beta.14",
```
Comment (fabioberger) [inline packages/base-contract/package.json]: Ditto above comment.

File: packages/contract-wrappers/package.json [hunk 3]
```diff
@@ -84,7 +84,7 @@
         "ethereum-types": "^1.0.8",
         "ethereumjs-blockstream": "6.0.0",
         "ethereumjs-util": "^5.1.1",
-        "ethers": "3.0.22",
+        "ethers": "^4.0.0-beta.14",
```
Comment (fabioberger) [inline packages/contract-wrappers/package.json]: ditto.

File: packages/contract_templates/contract.handlebars [hunk 4]
```diff
@@ -65,10 +65,12 @@ export class {{contractName}}Contract extends BaseContract {
             [{{> params inputs=ctor.inputs}}],
             BaseContract._bigNumberToString,
         );
-        const txData = ethers.Contract.getDeployTransaction(bytecode, abi, {{> params inputs=ctor.inputs}});
+        const iface = new ethers.Interface(abi);
```
Comment (fabioberger) [inline packages/contract_templates/contract.handlebars]: Let's just use variable name `interface`. Or better yet `contractInterface`

File: packages/contracts/package.json [hunk 5]
```diff
@@ -83,7 +83,7 @@
         "ethereum-types": "^1.0.8",
         "ethereumjs-abi": "0.6.5",
         "ethereumjs-util": "^5.1.1",
-        "ethers": "3.0.22",
+        "ethers": "^4.0.0-beta.14",
```
Comment (fabioberger) [inline packages/contracts/package.json]: ditto.

**Pull request merged**